create-quasar/templates/app/quasar-v2/ts-vite/BASE/src/components/EssentialLink.vue

```diff
@@ -19,8 +19,17 @@
   </q-item>
 </template>
 
-<script lang="ts">
-<% if (typescriptConfig === 'composition') { %>import { defineComponent } from 'vue';
+<% if (typescriptConfig === 'composition-setup') { %><script setup lang="ts">
+export interface EssentialLinkProps {
+  title: string;
+  caption?: string;
+  link?: string;
+  icon?: string;
+}
+const props = withDefaults(defineProps<EssentialLinkProps>(), {
+  link: '#',
+});<% } else if (typescriptConfig === 'composition') { %><script lang="ts">
```

Comment:
Changes to my previous suggestions:
`props` variable isn't used, so we should remove the assignment. Also, the vue3-recommended ruleset complains because we specify a default for some optional props and didn't for others. It would be a better idea to match the old structure anyway.

```suggestion
withDefaults(defineProps<EssentialLinkProps>(), {
  caption: '',
  link: '#',
  icon: '',
});<% } else if (typescriptConfig === 'composition') { %><script lang="ts">
```